File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/ui/Overview.java
package io.getstream.chat.docs.java.ui;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import io.getstream.chat.android.ui.feature.messages.list.MessageListView;
import io.getstream.chat.android.ui.viewmodel.messages.MessageListViewModel;
import io.getstream.chat.android.ui.viewmodel.messages.MessageListViewModelBinding;
import io.getstream.chat.android.ui.viewmodel.messages.MessageListViewModelFactory;

/**
 * [Overview](https://getstream.io/chat/docs/sdk/android/ui/overview/)
 */
public class Overview extends Fragment {

    private MessageListView messageListView;

    /**
     * [ViewModels](https://getstream.io/chat/docs/sdk/android/ui/overview/#viewmodels)
     */
    public void viewModels() {
        // 1
        ViewModelProvider.Factory factory = new MessageListViewModelFactory
                .Builder()
                .cid("messaging:123")
                .build();
        ViewModelProvider provider = new ViewModelProvider(this, factory);
        // 2
        MessageListViewModel viewModel = provider.get(MessageListViewModel.class);
        // 3
        MessageListViewModelBinding.bind(viewModel, messageListView, getViewLifecycleOwner());
    }
}

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/ui/Overview.kt
package io.getstream.chat.docs.kotlin.ui

import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import io.getstream.chat.android.ui.feature.messages.list.MessageListView
import io.getstream.chat.android.ui.viewmodel.messages.MessageListViewModel
import io.getstream.chat.android.ui.viewmodel.messages.MessageListViewModelFactory
import io.getstream.chat.android.ui.viewmodel.messages.bindView

/**
 * [Overview](https://getstream.io/chat/docs/sdk/android/ui/overview/)
 */
class Overview : Fragment() {

    private lateinit var messageListView: MessageListView

    /**
     * [ViewModels](https://getstream.io/chat/docs/sdk/android/ui/overview/#viewmodels)
     */
    fun viewModels() {
        // 1
        val factory = MessageListViewModelFactory(requireContext(), cid = "messaging:123")
        // 2
        val viewModel: MessageListViewModel by viewModels { factory }
        // 3
        viewModel.bindView(messageListView, viewLifecycleOwner)
    }
}
